### DISCUSSION TITLE
[New feature] Reddit stories

### DISCUSSION THREAD

**Discussion by MatteoFasulo** [MatteoFasulo/Whisper-TikTok#26] [Ideas]
I have an idea that I have seen in several places. In my case, I am creating Reddit stories, seen that they are being quite successful in those moments. And one feature that I see that everyone has is that, at the beginning of the video, a PNG appears as if it were a Reddit capture. Do you think you could make the script take the PNG fill it with text that has been put in the JSON, put it at the beginning of the video and hide the subtitles at the beginning

I also think it would be good if for example in the middle of the video you could add a subscribe video to encourage people to subscribe.

Thx

_Originally posted by @YotomiY in https://github.com/MatteoFasulo/Whisper-TikTok/discussions/13#discussioncomment-7965828_

**Comment by YotomiY** [comment]
Hello
Just want to explain clearly:
Like this:
![image](https://github.com/MatteoFasulo/Whisper-TikTok/assets/115656415/28838949-2a36-48a8-af61-f433a1998303)
This is also a great idea to replace the video.json method: At the webui we can ask "How many videos do you want to make?" so you answer (example: 2), "What Subreddit do you want to use?" (example: AskReddit) and "NFSW?" (example: no). Then using selenium (or the Reddit apps API) to extract the title of a post on the specific subreddit with or without nfsw and the content.
We put the title in a PNG template (like shown before, but blank ![image](https://github.com/MatteoFasulo/Whisper-TikTok/assets/115656415/57c11e4c-04ea-4c8a-99be-252439ad8a41) Found this one on internet)
Put the PNG at the start of the video and replace the"text" on the .json with the content

Also have to add another query : "Number of words or characters on the subtitles" (example: 1 word, 10 Characters)
I also think it would be good to add (optionally) a pup up animation to the subtitles
Thanks

**Comment by MatteoFasulo** [comment]
> Hello Just want to explain clearly: Like this: ![image](https://private-user-images.githubusercontent.com/115656415/294718286-28838949-2a36-48a8-af61-f433a1998303.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ3MzAyMTksIm5iZiI6MTcwNDcyOTkxOSwicGF0aCI6Ii8xMTU2NTY0MTUvMjk0NzE4Mjg2LTI4ODM4OTQ5LTJhMzYtNDhhOC1hZjYxLWY0MzNhMTk5ODMwMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEwOFQxNjA1MTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wMTcyNzI4MDNmMjJmMTFlZDVjMzVlZDBhMjhhZDAwZGVjNzkxNDMxZjkyYzBkYzI3MjhlMGNiOGY0ZWUwODY2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.aXTSuLpg4hKYUPIMc2bsjH73TroSvAVK84hUkE6MGSM) This is also a great idea to replace the video.json method: At the webui we can ask "How many videos do you want to make?" so you answer (example: 2), "What Subreddit do you want to use?" (example: AskReddit) and "NFSW?" (example: no). Then using selenium (or the Reddit apps API) to extract the title of a post on the specific subreddit with or without nfsw and the content. We put the title in a PNG template (like shown before, but blank ![image](https://private-user-images.githubusercontent.com/115656415/294718497-57c11e4c-04ea-4c8a-99be-252439ad8a41.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ3MzAyMTksIm5iZiI6MTcwNDcyOTkxOSwicGF0aCI6Ii8xMTU2NTY0MTUvMjk0NzE4NDk3LTU3YzExZTRjLTA0ZWEtNGM4YS05OWJlLTI1MjQzOWFkOGE0MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEwOFQxNjA1MTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iZGM0YmYxOTM3Y2RmNjQ5YzBkOWZmNTBjNmU2OTE1MjkyMDNlYmQzNDFlZjlhMmVjNzUxM2FkYjhlMzdjYWZjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.KlARq9SvnFji12QtK7P5x5gmj4W1CMXj-oxbm2Ml2YU) Found this one on internet) Put the PNG at the start of the video and replace the"text" on the .json with the content
> 
> Also have to add another query : "Number of words or characters on the subtitles" (example: 1 word, 10 Characters) I also think it would be good to add (optionally) a pup up animation to the subtitles Thanks

Hi there,

Thank you for your prompt response. I think it's a great idea to have a submodule for the Reddit-specific functionality within the repository. This way, it maintains modularity, and users interested in that specific feature can easily access and incorporate it without affecting the core functionality.

Before I start working on it, I wanted to check if you have any additional insights or if you think there's a better way to implement this Reddit functionality without making significant modifications to the entire software. Your input will be valuable as I dive into the development process.

Looking forward to your thoughts, and I'll ensure to keep you updated on the progress.

Best regards,
Matteo

**Comment by YotomiY** [comment]
Hello,
Thank you for your answer, For me this functionality could be optional in the gui, for example a checkbox to activate the "reddit" module. When you click, there will be several suboptions with the questions I mentioned the other day (What Subreddit do you want to use? ...). I don't know if it would be possible to make the functionality optional, but it would be cool to have 2 functionalities:
```
Video.json content, 
Reddit content, 
```
If it is not possible to create the functionality optionally, then just create a fork or a new branch

Draft here:
![Draft new feature Whisper-TikTok](https://github.com/MatteoFasulo/Whisper-TikTok/assets/115656415/25b3080e-295e-436f-8de8-59473e27ad5a)
Thank you

PS: 1 = Output video, and I'm having an error: when I create 2 or more videos it only shows 1 video, why not all of them?

**Comment by MatteoFasulo** [comment]
> Hello, Thank you for your answer, For me this functionality could be optional in the gui, for example a checkbox to activate the "reddit" module. When you click, there will be several suboptions with the questions I mentioned the other day (What Subreddit do you want to use? ...). I don't know if it would be possible to make the functionality optional, but it would be cool to have 2 functionalities:
> 
> ```
> Video.json content, 
> Reddit content, 
> ```
> 
> If it is not possible to create the functionality optionally, then just create a fork or a new branch
> 
> Draft here: ![Draft new feature Whisper-TikTok](https://private-user-images.githubusercontent.com/115656415/295010133-25b3080e-295e-436f-8de8-59473e27ad5a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ4ODQxMDcsIm5iZiI6MTcwNDg4MzgwNywicGF0aCI6Ii8xMTU2NTY0MTUvMjk1MDEwMTMzLTI1YjMwODBlLTI5NWUtNDM2Zi04ZGU4LTU5NDczZTI3YWQ1YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDExMFQxMDUwMDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03NWY5NmY3ZDZmZTI4ZTc4YjM0ZTExNDQ3M2U2MWJlM2FhZWE2NmU5YWYzZThjYWU5YzMxZDBlNGM2YTcxNmE1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.qVzcN-4oeFX8WDLhNM3QdT011xyHcdQPTXc5rKGgYqs) Thank you
> 
> PS: 1 = Output video, and I'm having an error: when I create 2 or more videos it only shows 1 video, why not all of them?

Hi,
there is only one video even if you generate more videos since Gradio interface expects a single component as output (which in the multiple case will be the last content generated). I'm not so skilled in working with Gradio so be patient while I try to figure out how to do that

Thx

**Comment by YotomiY** [comment]
Ok thanks, and for the reddit fonction?

**Comment by MatteoFasulo** [comment]
> Ok thanks, and for the reddit fonction?
> 
> 

I'm working on it :)

**Comment by MatteoFasulo** [comment]
I will also drop Gradio as a WebUI since I am not so confident in working on it. I'll try something with streamlit these days (there is already a version right now).

It should be already visible here: https://convert.streamlit.app/

If you can, test the new dashboard and feel free to make suggestions about it 👍🏼 

The reddit stuff will be on the page `reddit` visible in the sidebar. Still working on it but was not so confident with Gradio env.

**Comment by YotomiY** [comment]
Hello, 
How's it going? I am contacting you to tell you if you could inform me about how the new Reddit feature is going. You can also tell me if you need help.
Thanks!

**Reply by YotomiY** [reply]
So?

**Reply by MatteoFasulo** [reply]
Working on the UI part rn. Will commit asap

**Reply by MatteoFasulo** [reply]
The tricky part is getting the comments and put together the title of the submission and the comments

**Reply by MatteoFasulo** [reply]
Pushed the subreddit page (right now it is just scraping a random submission of the chosen subreddit and get the 10 most recent comments)

**Reply by YotomiY** [reply]
Excited to see the final feature @MatteoFasulo !!
Tell me if you have any problems and if you need it, this is the Reddit post blank template
Edit: Just notice that i already sennd it to you 😅
![image](https://github.com/MatteoFasulo/Whisper-TikTok/assets/115656415/15afbd58-f8d1-485e-9af6-1feae8bcf419)

**Reply by MatteoFasulo** [reply]
Hi,
thx for the template, btw I need to figure it out how to overlap the submission title on this template (not sure but it will require additional dependencies to the project). There is a [similar project](https://github.com/elebumm/RedditVideoMakerBot) which uses a different set of tools indeed.

**Reply by YotomiY** [reply]
Hi,
I have tested the other project but it has a lot of bugs and compatibility problems when it generates the ai voices. 
For the new dependencies, it could be cool to start creating an .bat (for local user. I will do it today) that creates an cuda or an venv python environment with python 10 or 11, after install the dependencies and launch the ui. Don't hesitate to create a new branch (Reddit) with the minimal advances in the feature so I can help you.
Bye 👋 
Thx

**Reply by MatteoFasulo** [reply]
Hi,
I will create a new branch "reddit". I'm working on the docker image in order to avoid creating several setup scripts :)

**Reply by YotomiY** [reply]
Ok, cool!!

**Reply by MatteoFasulo** [reply]
"reddit" branch will be an adaptation of the work of @duozokker for the WebUI version

**Comment by duozokker** [comment]
@YotomiY i made something just shy of it. Created a pull request and it got accepted. You can check it out near the end of the readme file where everything is neatly explained :D
